pages/u/[username]/voiseys/[voisey_id].tsx

```diff
@@ -65,7 +78,26 @@ const User: React.FC<UserProps> = ({ user: { username, photoUrl } }) => {
   );
 };
 
-const VoiseyPage: NextPage<Props> = ({ voisey, beat, host }) => {
+const VoiseyPageLoader: NextPage<LoaderProps> = ({
+  voisey,
+  beat,
+  host,
+  statusCode,
+}) => {
+  if (
+    typeof voisey !== 'undefined' &&
+    typeof beat !== 'undefined' &&
+    statusCode === 200
+  ) {
+    return <VoiseyPage voisey={voisey} beat={beat} host={host} />;
+  } else if (statusCode === 404) {
+    return <ErrorPage>Could not find voisey.</ErrorPage>;
+  } else {
+    return <Error statusCode={statusCode ?? 503} />;
+  }
+};
+
+const VoiseyPage = ({ voisey, beat, host }: PageProps) => {
```

Comment:
This is fine, but it's a little nicer to define the type of this as a react component with `PageProps` props:

```suggestion
const VoiseyPage: React.FC<PageProps> = ({ voisey, beat, host }) => {
```